coursetable/coursetable
# feat: add, remove, rename worksheets

Add, remove, rename worksheets. New logic to handle friends and currently viewed worksheets.
- Temporary API changes removed
- New UI for selecting worksheets/renaming/adding/deleting
- Main worksheet exists in DB iff it has nonzero courses
- Refactors, differentiating between "my viewed worksheet number" and "viewed worksheet number". Former is used for the one-click add, and latter is used for viewing purposes

TODO: Name of worksheet overflows onto buttons if too long

Fix https://github.com/coursetable/coursetable/issues/1457

Testing:
- Add worksheet
- Add worksheet with same name as curr worksheeet
- Add worksheet with "Main Worksheet"

- Rename not currently selected worksheet
- Rename currently selected worksheet

- Delete currently selected worksheet
- Delete not currently selected worksheet

- Select main worksheet, switch to friend, switch to one of their non-main worksheets, check still adding to main worksheet
- Select non main worksheet, switch to friend, switch to one of their non-main worksheets, check still adding to earlier selected non-main worksheet

- Course modal WS dropdown - selecting one of those still adds to correct worksheet
- Course modal WS dropdown - selecting one of those doesn't change the myCurrentlyViewedWS

- Change season - reverts to main worksheet, everything adds correctly, etc
- Adding course from friend's worksheet - adds to myCurrentlyViewedWorksheet
- Clicking out of friend's worksheets back to "me" - toggles to myCurrentlyViewedWorksheet
- Friend deletes worksheet that you're currently viewing, and you refresh - go to their main worksheet
- Remove course while viewing friend's worksheet that you both have - removes correctly from your worksheet, not theirs
